### PR TITLE
Created Docs-node-reuse.md as the guide for node reuse in hamilton library

### PR DESCRIPTION
**Pull request by danieltayal07**
Documentation for guide of node reuse in hamilton library

--- PR TEMPLATE INSTRUCTIONS (1) ---

Looking to submit a Hamilton Dataflow to the sf-hamilton-contrib module? If so go the the `Preview` tab and select the appropriate sub-template:
* [sf-hamilton-contrib template](?expand=1&template=HAMILTON_CONTRIB_PR_TEMPLATE.md)

Else, if not, please remove this block of text.

--- PR TEMPLATE INSTRUCTIONS (2) ---

[Short description explaining the high-level reason for the pull request]

## Changes

made a md file for the guide of node reuse

## How I tested this

i tested it by asking chat gpt to help me test it in my computer and then tested it in my computer

## Notes

tried my best to make the shortest and the most concise to the point guide for node reuse. 
Personally i never used hamilton library but now after seeing all the md's and the codes and how it makes out lives easier.
I am a proud user.

## Checklist

- [✅ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [✅ ] Changes are limited to a single goal (no scope creep)
- [ ✅] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ✅] Any _change_ in functionality is tested
- [ ✅] New functions are documented (with a description, list of inputs, and expected output)
- [✅ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ✅] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

**Comment by danieltayal07** [comment]
please add 
hacktoberfest2024
and hacktoberfest tag in this so it is counted in hacktoberfest2024

thank you!

**Review by skrawcz — CHANGES_REQUESTED** [review]
@danieltayal07 please see the #1045 issue. This doesn't fulfill those requirements and is not even correct hamilton code.

**Comment by danieltayal07** [comment]
Sorry for the mistake

I'll fix them and follow your requirements by the end of the day

**Comment by danieltayal07** [comment]
Hi @skrawcz I have tried to make better changes now, please check 

if not helpful please give feedback i'll try my best to mould it!

Thanks!

File: docs/how-tos/guide-reuse-nodes [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    # Function that calculates weekly sales using daily sales
+    def weekly_sales(daily_sales):
+        return daily_sales * 7
+    
+    # Function that calculates monthly sales using daily sales
+    def monthly_sales(daily_sales):
+        return daily_sales * 30
+    
+    # Create a Hamilton graph
+    graph = Graph()
```
Comment (elijahbenizzy) [inline docs/how-tos/guide-reuse-nodes:71]: What framework is this using? Very curious, I'm not familiar with this API.

**Review by skrawcz — CHANGES_REQUESTED** [review]
@danieltayal07 sorry but you're actually going to have to understand how Hamilton works to know whether the documentation is valid. Only using an LLM to generate this documentation isn't sufficient for this task.

**Comment by danieltayal07** [comment]
Yes i will try to find something easier than this to contribute to in Hamilton if possible, i apologise.

**Comment by skrawcz** [comment]
@danieltayal07 I have this issue which I think is perfect for you - https://github.com/DAGWorks-Inc/burr/issues/404 (it's in Hamilton's sister repo). Please comment if you'd like it.

**Comment by danieltayal07** [comment]
yes yes i would love to!!

**Pull request closed without merging**